mesonbuild/environment.py

```diff
@@ -1285,7 +1256,25 @@ def libdir_layout_is_win(self):
         return self.is_windows() \
             or self.is_cygwin()
 
-class MachineInfos(PerMachine):
+    # TODO make this compare two `MachineInfo`s purely. How important is the
+    # `detect_cpu_family({})` distinction? It is the one impediment to that.
+    def can_run(self):
+        """Whether we can run binaries for this machine on the current machine.
+
+        Can almost always run 32-bit binaries on 64-bit natively if the host
+        and build systems are the same. We don't pass any compilers to
+        detect_cpu_family() here because we always want to know the OS
+        architecture, not what the compiler environment tells us.
+        """
+        if self.system != detect_system():
+            return False
+        true_build_cpu_family = detect_cpu_family({})
+        return ((self.cpu_family == true_build_cpu_family) or
```

Comment:
__[Flake8]__ <sub>posted by Sider</sub>

[W504] line break after binary operator

Reply:
Guess I better go back to backslashes?